server/lib/nl/detection/place.py

```diff
@@ -48,93 +48,56 @@ def detect_from_query_dc(orig_query: str,
   # Recognize Places uses comma as a signal for contained-in-place.
   query = utils.remove_punctuations(orig_query, include_comma=True)
 
-  query_items = dc.recognize_places(query)
-
-  mains = []
-  main2corrections = {}
-  # List of parts in the query
-  query_parts: List[QueryPart] = []
-
   debug_logs["place_dcid_inference"] = {}
   debug_logs["place_resolution"] = {}
   debug_logs["dc_recognize_places"] = {}
 
-  for item in query_items:
-    if 'span' not in item:
-      continue
-    if 'places' in item and item['places'] and 'dcid' in item['places'][0]:
-      # Use the first DCID for now.
-      main = item['places'][0]['dcid']
-      mains.append(main)
-      query_parts.append(QueryPart(substr=item['span'].lower(), dcid=main))
-
-      related = []
-      for rp in item['places'][1:MAX_IDENTICAL_NAME_PLACES + 1]:
-        if 'dcid' in rp:
-          related.append(rp['dcid'])
-      main2corrections[main] = related
-
-      # For logging, get all DCIDs:
-      debug_logs["dc_recognize_places"][item['span']] = [
-          d['dcid'] for d in item['places'] if 'dcid' in d
-      ]
-    else:
-      query_parts.append(QueryPart(substr=item['span'].lower()))
-
-  resolved_places = []
-  resolved_entities = []
-  parent_map = {}
-  if mains:
-    resolved_places, parent_map = get_place_from_dcids(
-        mains, debug_logs["place_resolution"])
-    if allow_triples:
-      resolved_entities = _get_non_place_entities(mains, resolved_places)
-  resolved_place_dcids = set([p.dcid for p in resolved_places])
-  resolved_entity_dcids = set([e.dcid for e in resolved_entities])
+  place_query_parts = _get_query_parts(query, debug_logs["dc_recognize_places"],
+                                       True)
+  # Use the first DCID for now.
+  place_main_dcids = [p.dcids[0] for p in place_query_parts if p.dcids]
+  resolved_places, parent_map = get_place_from_dcids(
```

Comment:
For consistency, update other callsites of `get_place_from_dcids` to not check empty before calling?

Reply:
done